src/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -48,35 +50,57 @@ internal AuthenticationResult(AccessTokenCacheItem accessTokenCacheItem, Request
         /// <summary>
         /// Gets the Access Token requested.
         /// </summary>
-        public string AccessToken => _accessTokenCacheItem.AccessToken;
+        [DataMember]
+        public string AccessToken
+        {
+            get { return _accessTokenCacheItem.AccessToken; }
+            set =>  _accessTokenCacheItem.AccessToken = value; 
```

Comment:
please do not change any contracts on the public methods. there should be no setter property on any of the attributes in the class.